environ_functions.c

```diff
@@ -0,0 +1,153 @@
+#include "shell.h"
+
+/**
+ * print_list - Prints the key-value pairs of the linked list.
+ * @head: The head of the linked list.
+ */
+
+void print_list(Node *head)
+{
+	Node *current;
+
+	current = head;
+
+	while (current != NULL)
+	{
+		_puts(current->key);
+		write(1, "=", 1);
+		_puts(current->value);
+		write(1, "\n", 1);
+
+		current = current->next;
+	}
+}
+
+/**
+ * add_Node - Adds a new node with the specified key and value
+ * to the end of the linked list.
+ * @head: The head of the linked list.
+ * @key: The key to be added.
+ * @value: The value associated with the key.
+ * Return: The updated head of the linked list.
+ */
+
+Node *add_Node(Node *head, char *key, char *value)
+{
+	Node *newNode;
+	Node *current;
+
+	newNode = malloc(sizeof(Node));
+	if (newNode == NULL)
+	{
+		perror("Error: Memory allocation failed");
+		exit(EXIT_FAILURE);
```

Comment:
we should not exit from the program, just report the error and we will display the error in the shell.
the program shouldn't exit until the user exits from the shell